mllib/src/main/scala/org/apache/spark/ml/fpm/FPGrowth.scala

```diff
@@ -319,9 +323,11 @@ object FPGrowthModel extends MLReadable[FPGrowthModel] {
 
     override def load(path: String): FPGrowthModel = {
       val metadata = DefaultParamsReader.loadMetadata(path, sc, className)
+      implicit val format = DefaultFormats
+      val numTrainingRecords = (metadata.metadata \ "numTrainingRecords").extract[Long]
```

Comment:
Does this break backward compatibility for loading?

Reply:
Yes it does now. If saving numTrainingRecords to FPGrowthModel looks good, I can update the load logic.

Reply:
Since we're adding numTrainingRecords to FPGrowthModel and there isn't a proper default number, I suggest we break the backward model loading compatibility. Otherwise we need to fill numTrainingRecords with an incorrect value and will likely create a maintenance trap.